Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp

```diff
@@ -604,202 +604,215 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_string_tag_getter)
-        if (checked_target_byte_index.has_overflow())
-            return vm.throw_completion<RangeError>(global_object, "Overflow in target byte index");
-        auto target_byte_index = checked_target_byte_index.value();
-
-        // 23. Let limit be targetByteIndex + targetElementSize × srcLength.
-        Checked<size_t> checked_limit(source_length);
-        checked_limit *= target_element_size;
-        checked_limit += target_byte_index;
-        if (checked_limit.has_overflow())
-            return vm.throw_completion<RangeError>(global_object, "Overflow in target limit");
-        auto limit = checked_limit.value();
-
-        // 24. If srcType is the same as targetType, then
-        if (source_typed_array.element_name() == typed_array->element_name()) {
-            // a. NOTE: If srcType and targetType are the same, the transfer must be performed in a manner that preserves the bit-level encoding of the source data.
-            // b. Repeat, while targetByteIndex < limit,
-            //     i. Let value be GetValueFromBuffer(srcBuffer, srcByteIndex, Uint8, true, Unordered).
-            //     ii. Perform SetValueInBuffer(targetBuffer, targetByteIndex, Uint8, value, true, Unordered).
-            //     iii. Set srcByteIndex to srcByteIndex + 1.
-            //     iv. Set targetByteIndex to targetByteIndex + 1.
-            target_buffer->buffer().overwrite(target_byte_index, source_buffer->buffer().data() + source_byte_index, limit - target_byte_index);
-        } else {
-            // a. Repeat, while targetByteIndex < limit,
-            while (target_byte_index < limit) {
-                // i. Let value be GetValueFromBuffer(srcBuffer, srcByteIndex, srcType, true, Unordered).
-                auto value = source_typed_array.get_value_from_buffer(source_byte_index, ArrayBuffer::Unordered);
-                // ii. Perform SetValueInBuffer(targetBuffer, targetByteIndex, targetType, value, true, Unordered).
-                typed_array->set_value_in_buffer(target_byte_index, value, ArrayBuffer::Unordered);
-                // iii. Set srcByteIndex to srcByteIndex + srcElementSize.
-                source_byte_index += source_element_size;
-                // iv. Set targetByteIndex to targetByteIndex + targetElementSize.
-                target_byte_index += typed_array->element_size();
-            }
-        }
-    }
-    // 7. Else,
-    else {
-        // a. Perform ? SetTypedArrayFromArrayLike(target, targetOffset, source).
+    // 5. Let srcLength be ? LengthOfArrayLike(src).
+    auto source_length = TRY(length_of_array_like(global_object, *src));
 
-        // 23.2.3.23.2 SetTypedArrayFromArrayLike ( target, targetOffset, source ), https://tc39.es/ecma262/#sec-settypedarrayfromarraylike
+    // 6. If targetOffset is +∞, throw a RangeError exception.
+    if (isinf(target_offset))
+        return vm.throw_completion<RangeError>(global_object, "Invalid target offset");
```

Comment:
Let's add an `ErrorType` for this while we're here :)

Reply:
I ended up just adding a separate commit to do that here and in other places in the prototype file - there were a couple other places using this same error message